src/pe_source/data/shodan/shodan_search.py

```diff
@@ -0,0 +1,485 @@
+        risk_data,
+        columns=[
+            "organizations_uid",
+            "organization",
+            "ip",
+            "port",
+            "protocol",
+            "type",
+            "name",
+            "potential_vulns",
+            "mitigation",
+            "timestamp",
+            "product",
+            "server",
+            "tags",
+            "domains",
+            "hostnames",
+            "isn",
+            "asn",
+        ],
+    )
+    vuln_df = pd.DataFrame(
+        vuln_data,
+        columns=[
+            "organizations_uid",
+            "organization",
+            "ip",
+            "port",
+            "protocol",
+            "timestamp",
+            "cve",
+            "severity",
+            "cvss",
+            "summary",
+            "product",
+            "attack_vector",
+            "av_description",
+            "attack_complexity",
+            "ac_description",
+            "confidentiality_impact",
+            "ci_description",
+            "integrity_impact",
+            "ii_Description",
+            "availability_impact",
+            "ai_description",
+            "tags",
+            "domains",
+            "hostnames",
+            "isn",
+            "asn",
+        ],
+    )
+
+    # Grab the data source uid and add to each dataframe
+    source_uid = get_data_source_uid("Shodan")
+    df["data_source_uid"] = source_uid
+    risk_df["data_source_uid"] = source_uid
+    vuln_df["data_source_uid"] = source_uid
+
+    # Insert data into the PE database
+    failed = insert_shodan_data(df, "shodan_assets", thread_name, org_name, failed)
+    failed = insert_shodan_data(
+        risk_df,
+        "shodan_insecure_protocols_unverified_vulns",
+        thread_name,
+        org_name,
+        failed,
+    )
+    failed = insert_shodan_data(
+        vuln_df, "shodan_verified_vulns", thread_name, org_name, failed
+    )
+
+    return failed
+
+
+def is_verified(
+    vulns, cve, av_dict, ac_dict, ci_dict, vuln_data, org_uid, r, d, asn, unverified
+):
+    """Check if a CVE is verified."""
+    v = vulns[cve]
+    if v["verified"]:
+        re = search_circl(cve)
+        r_json = re.json()
+        if r_json is not None:
+            summary = r_json.get("summary", None)
+            product = r_json.get("vulnerable_product", None)
+            attack_vector = r_json.get("access", {}).get("vector")
+            av = av_dict.get(attack_vector, None)
+            attack_complexity = r_json.get("access", {}).get("complexity")
+            ac = ac_dict.get(attack_complexity, None)
+            conf_imp = r_json.get("impact", {}).get("confidentiality")
+            ci = ci_dict.get(conf_imp, None)
+            int_imp = r_json.get("impact", {}).get("integrity")
+            ii = ci_dict.get(int_imp, None)
+            avail_imp = r_json.get("impact", {}).get("availability")
+            ai = ci_dict.get(avail_imp, None)
+            cvss = r_json.get("cvss", None)
```

Comment:
No need to specify `None` here, since that is the default return value of `get()` if the specified key is not present in the dictionary.
```suggestion
            summary = r_json.get("summary")
            product = r_json.get("vulnerable_product")
            attack_vector = r_json.get("access", {}).get("vector")
            av = av_dict.get(attack_vector)
            attack_complexity = r_json.get("access", {}).get("complexity")
            ac = ac_dict.get(attack_complexity)
            conf_imp = r_json.get("impact", {}).get("confidentiality")
            ci = ci_dict.get(conf_imp)
            int_imp = r_json.get("impact", {}).get("integrity")
            ii = ci_dict.get(int_imp)
            avail_imp = r_json.get("impact", {}).get("availability")
            ai = ci_dict.get(avail_imp)
            cvss = r_json.get("cvss")
```